commcare_connect/opportunity/utils/completed_work.py

```diff
@@ -35,3 +41,16 @@ def update_status(completed_works, opportunity_access, compute_payment=True):
     if compute_payment:
         opportunity_access.payment_accrued = payment_accrued
         opportunity_access.save()
+
+
+def update_work_payment_date(payment: Payment):
+    completed_works = CompletedWork.objects.filter(opportunity_access=payment.opportunity_access)
```

Comment:
These need to be ordered by approval date (probably `status_modified_date` is fine), and exclude ones that are already marked paid (otherwise, we will just keep updating the first X).

Reply:
[802e47b](https://github.com/dimagi/commcare-connect/pull/405/commits/802e47bfa002fde581ce5067751a2f4e60cc059d)